proposals/0021-root-signature-in-clang.md

```diff
@@ -0,0 +1,239 @@
+<!-- {% raw %} -->
+
+# Implementation of RootSignature in clang
+
+* Proposal: [0021](0021-root-signature-in-clang.md)
+* Author(s): [Xiang Li](https://github.com/python3kgae)
+* Sponsor: TBD
+* Status: **Under Consideration**
+
+
+## Introduction
+
+In HLSL, the [root signature](https://learn.microsoft.com/en-us/windows/win32/direct3d12/root-signatures)
+defines what types of resources are bound to the graphics pipeline.
+
+A root signature can be specified in HLSL as a [string](https://learn.microsoft.com/en-us/windows/win32/direct3d12/specifying-root-signatures-in-hlsl#an-example-hlsl-root-signature).
+The string contains a collection of comma-separated clauses that describe root
+signature constituent components.
+
+## Motivation
+
+To support root signature in clang-dxc.
+
+## Requirements
+
+1) Clang-dxc should parse existing root signature grammars and provide 
```

Comment:
nit: Requirements aren't "should" sentences, they should be "must" or "shall". I realize this is a bit wishy-washy language, but "should" doesn't state a requirement it states a nice-to-have.

Reply:
Updated